### ISSUE TITLE
Pythonize regridding atmospheric forcing (original ncl, esmf)

### ISSUE THREAD

**Issue by ziu1986** [NorESMhub/NorESM_LandSites_Platform#16]
Existing ncl scripts by @huitang-earth have to be pythonized using esmpy wrapper for ESMF.

**Comment by ziu1986** [comment]
I had a little spare time to think about the issue. 

- ESMF integration is probably the most flexible option we have. But there are indications that the python interface might not provide all functionalities we would need. I cannot judge this at this point. @huitang-earth You have worked with ESMF previously. You may have a look at the "Limitations" here: https://earthsystemmodeling.org/esmpy_doc/release/ESMF_8_0_1/html/install.html
- In case we only want to provide extraction of atmospheric forcing data, e.g., for single point simulations, a simple 4-D vector algebra might do the trick. Mathematically, we should be able to do the interpolation in all 4 dimensions simultaneously. However, the simple algorithm I have in mind grows exponentially in computational cost with dimension.

Basically:
Our variable for interpolation to the site level: m_s
Its local vector in spherical coordinates: vec_r_s = (t_s, r_s, phi_s, theta_s)
4-16 stenciles: vec_r_j^i = (t_j, r_j^i, phy_j^i, theta_j^i)
with variable values: m_j^i

m_s = sum(|vec_r_s - vec_r_j^i| x m_j^i) / sum(|vec_r_s - vec_r_j^i|)

**Comment by emiliano-gelati** [comment]
For interpolation and re-gridding, the Python [xarray](http://xarray.pydata.org/en/stable/) package offers some basic tools:
http://xarray.pydata.org/en/stable/interpolation.html
http://xarray.pydata.org/en/stable/generated/xarray.DataArray.interp.html

**Comment by evalieungh** [comment]
Is this still relevant? Closing the issue for now under the assumption we have solved this in some other way.
